lib/services/product-manager.js

```diff
@@ -202,6 +203,65 @@ export default class ProductManager {
       : Promise.resolve(product)
   }
 
+  /**
+   * Method will take a product with variants which should be anonymized.
+   * It will modify slug, key and name so the product can be created without
+   * issues with duplicate fields
+   * @param product productDraft
+   */
+  getAnonymizedProduct (product) {
+    const timestamp = this._getTimestamp()
+
+    // make all languages in slug unique
+    for (const lang in product.slug) // eslint-disable-line guard-for-in
+      product.slug[lang] += `-${timestamp}`
+
+    if (product.key)
+      product.key += `-${timestamp}`
+    product.slug[constant.PRODUCT_ANONYMIZE_SLUG_KEY] = timestamp
+
+    return product
+  }
+
+  /**
+   * Method will take a product with current and staged versions
+   * it will unpublish it and update slugs so they are unique across CTP project
+   * @param product ctpProduct
+   */
+  anonymizeCtpProduct (product) {
+    const timestamp = this._getTimestamp()
+    const actions = []
+
+
+    if (product.masterData.published)
+      actions.push({
+        action: 'unpublish'
```

Comment:
Will this help? If you unpublish, the product is still in current, but the only thing that changes is isPublished attribute, right?

Reply:
@lojzatran so you think we should not unpublish product if we change its slug and key?

Reply:
I'm not sure. Did you try to see if this does what you expected?

Reply:
Unpublish action will just remove the product from catalog of published products so the question is if this updated product should stay in the eshop or if we should remove it from there..

Reply:
Let's leave it as you have it here.